indexing-service/src/main/java/io/druid/indexing/worker/WorkerTaskMonitor.java

```diff
@@ -237,18 +242,21 @@ public void onFailure(Throwable t)
   }
 
   @LifecycleStop
-  public void stop()
+  public void stop() throws InterruptedException
   {
     synchronized (lifecycleLock) {
       Preconditions.checkState(started, "not started");
 
       try {
+        log.info("Stopping WorkerTaskMonitor.");
+        started = false;
         exec.shutdownNow();
         pathChildrenCache.close();
         taskRunner.stop();
-
-        started = false;
-        log.info("Stopped WorkerTaskMonitor.");
+        doneStopping.await();
```

Comment:
It doesn't, do you think it should? should the value be configurable?

My attitude is generally that program-lifecycle stop() things don't need to time out, as people should have some higher level thing that will kill -9 the proc if it takes too long to shut down. But we can have stop() timeouts if we want.